RawSpeed/ArwDecoder.cpp

```diff
@@ -272,6 +306,88 @@ void ArwDecoder::decodeMetaDataInternal(CameraMetaData *meta) {
   mRaw->blackLevel >>= mShiftDownScale;
 }
 
+void ArwDecoder::GetWB() {
+  // Set the whitebalance
+  if (mRootIFD->hasEntryRecursive(DNGPRIVATEDATA)) {
+    TiffEntry *priv = mRootIFD->getEntryRecursive(DNGPRIVATEDATA);
+    const uchar8 *data = priv->getData();
+    uint32 off = (uint32)data[3] << 24 | (uint32)data[2] << 16 | (uint32)data[1] << 8 | (uint32)data[0];
+    TiffIFD *sony_private;
+    if (mRootIFD->endian == getHostEndianness())
+      sony_private = new TiffIFD(mFile, off);
+    else
+      sony_private = new TiffIFDBE(mFile, off);
+
+    TiffEntry *sony_offset = sony_private->getEntryRecursive(SONY_OFFSET);
+    TiffEntry *sony_length = sony_private->getEntryRecursive(SONY_LENGTH);
+    TiffEntry *sony_key = sony_private->getEntryRecursive(SONY_KEY);
+    if(!sony_offset || !sony_length || !sony_key || sony_key->count != 4)
+      ThrowRDE("ARW: couldn't find the correct metadata for WB decoding");
```

Comment:
We shouldn't abort decoding just because we cannot read WB data, so these exceptions should be caught.